main.py

```diff
@@ -1,42 +1,52 @@
+import pandas as pd
 from games import Games
-from players import Players
+from player_comparator import PlayerComparator
 from play_by_play import PlayByPlay
-from connect import Connector
-from executor import Executor
+from nba_api.stats.endpoints import playbyplayv2
+from models import engine, Players
+from sqlalchemy.orm import sessionmaker
+import time
 import datetime
 
-today = datetime.date.today()
-yesterday = today - datetime.timedelta(days=1)
-from_date = to_date = yesterday.isoformat()
-
-# Connect to DB and create a cursor for executing queries.
-connector = Connector()
-conn, cursor = connector.connect()
-
-# Create an executor instance for writing to DB
-exec = Executor(conn, cursor)
+from_date = to_date = (datetime.date.today() - datetime.timedelta(days=1)).strftime('%m/%d/%Y')
 
 # Retrieve Games
-yesterdays_games = Games(from_date, to_date)
-yesterdays_games_data = yesterdays_games.get_games()
-game_ids = list(yesterdays_games_data['GAME_ID'])
-
-# Retrieve play by play data from those games, sorted by game chronologically
-playbyplay = PlayByPlay(game_ids)
-plays = playbyplay.get_plays()
-
-# Extract players from plays data
-nba_players = Players()
-players_to_add = nba_players.compare_to_existing(plays)
-if players_to_add:
-    exec.execute_values(players_to_add, 'players')
-
-# Write games data to DB
-exec.execute_values(yesterdays_games_data, 'games')
-
-# Write play by play data to DB
-exec.execute_values(plays, 'playbyplay')
+print(f'Finding games for {from_date}.')
+game_finder = Games(from_date, to_date)
+games_df = game_finder.get_games()
+game_ids = list(games_df['game_id'])
+
+if game_ids:
```

Comment:
I'd recommend breaking up this `main` file into a few parts. 

1. Extract Data - Games and Play by Play
2. Process Data - Grouping the playby play with the player ids.
3. Write The Data - Write the resulting models to database.


I think the majority of the (2) can be in it's own file that can be tested. For example lines 19-40 should be in a separate file. 

My recommendation is to break this up into 3 files instead of just one `main.py`

1. `get_and_write_play_by_play_data_main.py` - replaces this file as the main function.
2. `get_and_write_play_by_play.py` - Does what's on lines 19-40. Accepts all of the raw data (games, play by play) as arguments to a class or function. Then returns the play by play data in the proper format. This can use your other helper classes. This class should not have any dependencies on SQL calls since the data comes in as objects. 
3. `get_and_write_play_by_play_test.py` - Unit tests for (2) these should be easier to do since there is no dependency on SQL. No need to mock them.